subsys/usb/device_next/usbd_core.c

```diff
@@ -128,33 +128,39 @@ static ALWAYS_INLINE int usbd_event_handler(struct usbd_contex *const uds_ctx,
 
 	switch (event->type) {
 	case UDC_EVT_VBUS_REMOVED:
-		LOG_WRN("VBUS remove event");
+		LOG_DBG("VBUS remove event");
+		usbd_msg_pub_simple(uds_ctx, event->type, 0);
 		break;
 	case UDC_EVT_VBUS_READY:
-		LOG_WRN("VBUS detected event");
+		LOG_DBG("VBUS detected event");
+		usbd_msg_pub_simple(uds_ctx, event->type, 0);
```

Comment:
Don't silently cast one enum to another! Use proper enum value here (`USBD_MSG_VBUS_READY`).

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/68256#discussion_r1474610952